qiskit_sphinx_theme/__init__.py

```diff
@@ -25,6 +27,35 @@ def _get_theme_absolute_path(folder_name: str) -> str:
     return str(path.resolve())
 
 
+def remove_thebe_if_not_needed(app, pagename, templatename, context, doctree):
+    """
+    Remove files that jupyter-sphinx incorrectly tries to add.
+    
+    See https://github.com/Qiskit/qiskit_sphinx_theme/issues/291 for more context. 
+    """
+    # jupyter-sphinx might be not installed. 
+    try:
+        from jupyter_sphinx.thebelab import ThebeButtonNode
+    except ImportError:
+        return 
+
+    if not doctree:
```

Comment:
It would be helpful to "refactor" this code here to "invert the conditional" and use "early returns". See https://medium.com/swlh/return-early-pattern-3d18a41bba8 for an explanation of what that means. It reduces the amount of nesting and makes your code easier to understand.

```suggestion
    if not doctree or doctree.traverse(ThebeButtonNode):
```

You would apply my suggestion above, and then remove line 45 (`if not doctree.traverse(ThebeButtonNode)`), then unindent everything in the if block.